components/core/src/os/process/linux.rs

```diff
@@ -32,8 +33,13 @@ pub fn become_command(command: PathBuf, args: Vec<OsString>) -> Result<()> {
 }
 
 /// send a Unix signal to a pid
-pub fn send_signal(pid: u32, sig: u32) -> u32 {
-    unsafe { kill(pid as i32, sig) }
+fn send_signal(pid: u32, sig: libc::c_int) -> Result<()> {
+    unsafe {
+        match kill(pid as i32, sig as u32) {
+            0 => Ok(()),
+            _ => return Err(Error::SignalFailed),
```

Comment:
We should give the non-zero error code to the `SignalFailed` error to help with debugging